src/libpgmoneta/configuration.c

```diff
@@ -3021,3 +3041,48 @@ is_empty_string(char* s)
 
    return true;
 }
+
+static int
+remove_leading_whitespace(const char* s, char** trimmed_line)
+{
+   // Find the index of the first non-whitespace character
+   int i = 0;
+   while (s[i] != '\0' && isspace(s[i]))
+   {
+      i++;
+   }
+
+   *trimmed_line = NULL; // Initialize trimmed_line
+
+   // Loop through the string starting from non-whitespace character
+   for (; s[i] != '\0'; i++)
+   {
+      *trimmed_line = pgmoneta_append_char(*trimmed_line, s[i]);
```

Comment:
Use a separate variable to assign, and then do `*trimmed_line = X` before returning